Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -1126,17 +1280,103 @@ - (int)maxMessageWidthForContentWidth:(int)contentWidth
     return (int)floor(contentWidth * 0.8f);
 }
 
+- (CGSize)quotedMessageSizeForViewWidth:(int)viewWidth
+                           contentWidth:(int)contentWidth
+                         includeMargins:(BOOL)includeMargins
+{
+    OWSAssert(self.viewItem);
+    OWSAssert([self.viewItem.interaction isKindOfClass:[TSMessage class]]);
+
+    CGSize result = CGSizeZero;
+
+    if (!self.isQuotedReply) {
+        return result;
+    }
+
+    result.width += self.quotedMessageHInset;
+    result.width += self.quotedReplyStripeThickness;
+    result.width += self.quotedReplyStripeHSpacing;
+
+    result.height += self.quotedMessageTopInset;
+
+    CGFloat thumbnailHeight = 0.f;
+    if (self.hasQuotedAttachmentThumbnail) {
+        result.width += self.quotedThumbnailHSpacing;
+        result.width += self.quotedThumbnailSize;
+
+        thumbnailHeight = self.quotedThumbnailSize;
+    } else {
+        result.width += self.quotedContentTrailingMargin;
+    }
+
+    result.width += self.quotedMessageHInset;
+
+    // Once we've determined everything _except_ the size of the text
+    // content (i.e. the quoted author and the quoted text (if any)),
+    // we can determine the size of the text content.
+    const int maxMessageWidth = [self maxMessageWidthForContentWidth:contentWidth];
+    CGFloat maxTextWidth = (maxMessageWidth - (self.textTrailingMargin + self.textLeadingMargin + result.width));
+    CGFloat textWidth = 0.f;
+
+    // Author
+    {
+        OWSContactsManager *contactsManager = Environment.current.contactsManager;
+        NSString *quotedAuthor = [contactsManager displayNameForPhoneIdentifier:self.viewItem.quotedRecipientId];
+
+        UILabel *quotedAuthorLabel = [UILabel new];
+        quotedAuthorLabel.text = quotedAuthor;
+        quotedAuthorLabel.font = self.quotedAuthorFont;
+        quotedAuthorLabel.lineBreakMode = NSLineBreakByTruncatingTail;
+        quotedAuthorLabel.numberOfLines = 1;
+
+        CGSize quotedAuthorSize = CGSizeCeil([quotedAuthorLabel sizeThatFits:CGSizeMake(maxTextWidth, CGFLOAT_MAX)]);
+
+        textWidth = MAX(textWidth, quotedAuthorSize.width);
+        result.height += self.quotedContentTopInset;
+        result.height += self.quotedAuthorHeight;
+    }
+
+    if (self.hasQuotedText) {
+        UILabel *quotedTextView = [self createQuotedTextView];
```

Comment:
Per above, this is potentially misleading because it looks like we could safely mutate quotedTextView here, but it's the same quotedTextView that's used when displaying the cell - we're not actually creating one.

Even though it's one more line, I think this is better:

```
[self resetQuotedTextView];
UILabel *quotedTextView = self.quotedTextView;
```